docs/tildagonos-apps/guides/ui-widgets.md

```diff
@@ -0,0 +1,89 @@
+# UI Widget Overview
+
+You can use the following premade [`app_components`](guides/ui-widgets.md) to create user interfaces:
+
+## Menu
+
+The [`Menu`](https://github.com/emfcamp/badge-2024-software/blob/main/modules/app_components/menu.py) component, allows you to create menus with which you can scroll through a list of options and select an item.
+
+## Notification
+
+The [`Notification`](https://github.com/emfcamp/badge-2024-software/blob/main/modules/app_components/notification.py) component allows you to create pop up notifications.
+
+### Example
+
+This example app shows the notification `down!` when you press the down button.
+
+```python
+from app import App
+from app_components import Notification, clear_background
+from events.input import Buttons, BUTTON_TYPES
+
+
+class NotificationDemo(App):
+    def __init__(self):
+        self.notification = None
+        self.button_states = Buttons(self)
+
+    def update(self, delta):
+        if self.button_states.get(BUTTON_TYPES["CANCEL"]):
+            self.minimise()
+        if self.button_states.get(BUTTON_TYPES["DOWN"]):
+            self.notification = Notification("down!")
+        if self.notification:
+            self.notification.update(delta)
+
+    def draw(self, ctx):
+        if self.notification:
+            self.notification.draw(ctx)
+
+```
+
+### Usage
+
+To use a notification:
+
+1. Import the `Notifiaction` component:
+
+    ```python
+    from app_components import Notification
+    ```
+
+2. Initialize the notification when in the `__init__` method of your app:
+
+    ```python
+    self.notification = None
+    ```
+
+3. Set the notification text in your app's code:
+
+    ```python
+    self.notification = Notification("This is a notification!")
+    ```
+
+4. Add the following lines in your `draw()` method to draw the notification when `self.notification` contains a notification:
```

Comment:
I feel like we should say more about the draw method too?

Reply:
The draw method needs to be snappy, since it's called for every frame. There's been some talk though about modifying it so that it tells the called functions which parts of the screen haven't been touched since the last draw, so that it can draw only the relevant parts of the screen - it seems like that'll be a good opportunity to go more into detail about performance requirements.